### ISSUE TITLE
All git fetch requests result in 404, repository not found

### ISSUE THREAD

**Issue by mkstephenson** [WeblateOrg/weblate#12011]
### Describe the issue

We are running Weblate in a dockerised environment using Azure AD for authentication and our upstream git repos are all cloned from Azure DevOps using SSH. We are now getting some merge conflicts after using the service for a while and would like to resolve these using the instructions in the documentation (https://docs.weblate.org/en/latest/faq.html#how-to-fix-merge-conflicts-in-translations), however all attempts to fetch the changes from the weblate repo result in a 404, repository not found. Authentication works fine, if we leave out our token or change it in any way, we get the 401, as expected.

Upon closer inspection, I also realised that the repositories themselves don't contain the exported repository URL in any of the configuration tabs for the component, I have a feeling this might be connected. Is there anyway of debugging if the git repos are available at all or not?

The git exporter has **not** been turned off (we recently learned that it is enabled by default in the dockerised environment, which is good for us).

### I already tried

- [X] I've read and searched [the documentation](https://docs.weblate.org/).
- [X] I've searched for similar filed issues in this repository.

### Steps to reproduce the behavior

Set up a dockerised version of weblate using the following environment file (secrets and concrete URLs have been redacted):

```
WEBLATE_DEBUG=1
#WEBLATE_LOGLEVEL=INFO
WEBLATE_SITE_TITLE="Translations"
WEBLATE_ADMIN_NAME=admin
WEBLATE_ADMIN_EMAIL=<redacted>
WEBLATE_ADMIN_PASSWORD=<redacted>
WEBLATE_SERVER_EMAIL=weblate@<redacted>
WEBLATE_DEFAULT_FROM_EMAIL=weblate@<redacted>
WEBLATE_ALLOWED_HOSTS=*
WEBLATE_REGISTRATION_OPEN=0
WEBLATE_SITE_DOMAIN=<redacted>.westeurope.cloudapp.azure.com
WEBLATE_AUTO_UPDATE=full
WEBLATE_IP_PROXY_HEADER=HTTP_X_FORWARDED_FOR
WEBLATE_ENABLE_HTTPS=1
WEBLATE_REQUIRE_LOGIN=1
WEBLATE_AZURE_DEVOPS_CREDENTIALS='{"dev.azure.com":{"username":"<redacted>","token":"<redacted>","organization":"<redacted>"}}'
WEBLATE_AUTHENTICATION_BACKENDS='["social_core.backends.azuread_tenant.AzureADTenantOAuth2","social_core.backends.email.EmailAuth","weblate.accounts.auth.WeblateUserBackend"]'
WEBLATE_REGISTRATION_ALLOW_BACKENDS=azuread-tenant-oauth2
WEBLATE_SOCIAL_AUTH_AZUREAD_TENANT_OAUTH2_KEY = "<redacted>"
WEBLATE_SOCIAL_AUTH_AZUREAD_TENANT_OAUTH2_SECRET = "<redacted>"
WEBLATE_SOCIAL_AUTH_AZUREAD_TENANT_OAUTH2_TENANT_ID = "<redacted>"
POSTGRES_PASSWORD=<redacted>
POSTGRES_USER=weblate
POSTGRES_DATABASE=weblate
POSTGRES_HOST=database
REDIS_HOST=cache
REDIS_PORT=6379
WEBLATE_EMAIL_BACKEND="django.core.mail.backends.console.EmailBackend"
```

docker-compose file for this setup:
```
services:
  weblate:
    image: weblate/weblate
    tmpfs:
      - /run
      - /tmp
    volumes:
      - type: bind
        source: /app/data
        target: /app/data
      - type: bind
        source: /app/cache
        target: /app/cache
      - type: bind
        source: /app/emails
        target: /app/emails
    env_file:
      - ./environment
    restart: always
    read_only: true
    depends_on:
      - database
      - cache
  database:
    image: postgres:16-alpine
    env_file:
      - ./environment
    volumes:
      - type: bind
        source: /app/postgres-data
        target: /var/lib/postgresql/data
    restart: always
  https-portal:
    image: steveltn/https-portal:1
    ports:
    - 80:80
    - 443:443
    restart: always
    env_file:
      - ./environment
    environment:
      STAGE: production
      PROXY_READ_TIMEOUT: 3600
    volumes:
      - type: bind
        source: /app/ssl-certs
        target: /var/lib/https-portal
  cache:
    image: redis:7-alpine
    restart: always
    read_only: true
    command: [redis-server, --save, "60", "1"]
    volumes:
      - type: bind
        source: /app/redis-data
        target: /data
```

- Create a project and mark it as private.
- Set up a component using Azure DevOps pull request as the version control system and the ssh URL in the format git@ssh.dev.azure.com:v3/<organisation>/<area>/<project> as the source code repository.
- Also observe that there is no exported repository URLs field available.
- Attempt to clone the component you just created using the instructions found at https://docs.weblate.org/en/latest/faq.html#how-to-fix-merge-conflicts-in-translations for an authenticated clone/fetch and observe the error: fatal: repository 'https://<weblate-server>/git/<project>/<component>/' not found.

### Expected behavior

We would expect a successful clone (or git fetch) for the component.

### How do you run Weblate?

Docker container

### Weblate versions

 * Weblate: 5.6.1
 * Django: 5.0.6
 * siphashc: 2.4.1
 * translate-toolkit: 3.13.1
 * lxml: 5.2.2
 * pillow: 10.3.0
 * nh3: 0.2.17
 * python-dateutil: 2.9.0.post0
 * social-auth-core: 4.5.4
 * social-auth-app-django: 5.4.1
 * django-crispy-forms: 2.2
 * oauthlib: 3.2.2
 * django_compressor: 4.5
 * djangorestframework: 3.15.2
 * django-filter: 24.2
 * django-appconf: 1.0.6
 * user-agents: 2.2.0
 * filelock: 3.15.4
 * rapidfuzz: 3.9.3
 * openpyxl: 3.1.4
 * celery: 5.4.0
 * django-celery-beat: 2.6.0
 * kombu: 5.3.7
 * translation-finder: 2.16
 * weblate-language-data: 2024.5
 * html2text: 2024.2.26
 * pycairo: 1.26.1
 * PyGObject: 3.48.2
 * diff-match-patch: 20230430
 * requests: 2.32.3
 * django-redis: 5.4.0
 * hiredis: 2.3.2
 * sentry-sdk: 2.6.0
 * Cython: 3.0.10
 * misaka: 2.1.1
 * GitPython: 3.1.43
 * borgbackup: 1.2.8
 * pyparsing: 3.1.2
 * ahocorasick_rs: 0.22.0
 * python-redis-lock: 4.0.0
 * charset-normalizer: 3.3.2
 * Python: 3.12.4
 * Git: 2.39.2
 * psycopg: 3.1.19
 * psycopg-binary: 3.1.19
 * phply: 1.2.6
 * ruamel.yaml: 0.18.6
 * tesserocr: 2.7.0
 * boto3: 1.34.132
 * zeep: 4.2.1
 * aeidon: 1.15
 * iniparse: 0.5
 * mysqlclient: 2.2.4
 * Mercurial: 6.7.4
 * git-svn: 2.39.2
 * git-review: 2.4.0
 * PostgreSQL server: 16.3
 * Database backends: django.db.backends.postgresql
 * PostgreSQL implementation: psycopg3 (binary)
 * Cache backends: default:RedisCache, avatar:FileBasedCache
 * Email setup: django.core.mail.backends.console.EmailBackend: localhost
 * OS encoding: filesystem=utf-8, default=utf-8
 * Celery: redis://cache:6379/1, redis://cache:6379/1, regular
 * Platform: Linux 6.5.0-1023-azure (x86_64)

### Weblate deploy checks

```shell
System check identified some issues:

WARNINGS:
?: (security.W018) You should not have DEBUG set to True in deployment.

INFOS:
?: (weblate.I021) Error collection is not set up, it is highly recommended for production use
        HINT: https://docs.weblate.org/en/weblate-5.6.1/admin/install.html#collecting-errors
?: (weblate.I028) Backups are not configured, it is highly recommended for production use
        HINT: https://docs.weblate.org/en/weblate-5.6.1/admin/backup.html
?: (weblate.I031) New Weblate version is available, please upgrade to 5.6.2.
        HINT: https://docs.weblate.org/en/weblate-5.6.1/admin/upgrade.html

System check identified 4 issues (1 silenced).
```

**Comment by nijel** [comment]
You will get 401 without authentication. When the token you've used doesn't have access to the repo, you will get 404, so maybe the token is not having access to the repo?

**Comment by mkstephenson** [comment]
The token should have full access to the repo. It was the token fetched directly from a user profile which has Administrator access for the entire project (I added the VCS role to the user as well, no difference). The same user can see the component and project in the UI.
I also tried with a token generated directly in the component (username bot-<project>-<component>, token copied from web page), with the same 404 result. This token also had Administrator and VCS roles.
